src/openrct2-ui/windows/TrackDesignPlace.cpp

```diff
@@ -70,633 +70,553 @@ static rct_widget window_track_place_widgets[] = {
     WIDGETS_END,
 };
 
-static void WindowTrackPlaceClose(rct_window *w);
-static void WindowTrackPlaceMouseup(rct_window *w, rct_widgetindex widgetIndex);
-static void WindowTrackPlaceUpdate(rct_window *w);
-static void WindowTrackPlaceToolupdate(rct_window* w, rct_widgetindex widgetIndex, const ScreenCoordsXY& screenCoords);
-static void WindowTrackPlaceTooldown(rct_window* w, rct_widgetindex widgetIndex, const ScreenCoordsXY& screenCoords);
-static void WindowTrackPlaceToolabort(rct_window *w, rct_widgetindex widgetIndex);
-static void WindowTrackPlaceUnknown14(rct_window *w);
-static void WindowTrackPlaceInvalidate(rct_window *w);
-static void WindowTrackPlacePaint(rct_window *w, rct_drawpixelinfo *dpi);
-
-static rct_window_event_list window_track_place_events([](auto& events)
-{
-    events.close = &WindowTrackPlaceClose;
-    events.mouse_up = &WindowTrackPlaceMouseup;
-    events.update = &WindowTrackPlaceUpdate;
-    events.tool_update = &WindowTrackPlaceToolupdate;
-    events.tool_down = &WindowTrackPlaceTooldown;
-    events.tool_abort = &WindowTrackPlaceToolabort;
-    events.viewport_rotate = &WindowTrackPlaceUnknown14;
-    events.invalidate = &WindowTrackPlaceInvalidate;
-    events.paint = &WindowTrackPlacePaint;
-});
 // clang-format on
 
 static std::vector<uint8_t> _window_track_place_mini_preview;
 static CoordsXY _windowTrackPlaceLast;
-
 static RideId _window_track_place_ride_index;
 static bool _window_track_place_last_was_valid;
-static CoordsXYZ _windowTrackPlaceLastValid;
 static money32 _window_track_place_last_cost;
+static CoordsXYZ _windowTrackPlaceLastValid;
```

Comment:
In general we are wanting to move all of these static variables into the window. You will need to modify the functions that use these directly from other places e.g. `TrackPlaceClearProvisionalTemporarily` and `TrackPlaceRestoreProvisional` make it so those functions find the track window with `window_find_by_class` and then call a public function.

```cpp
void TrackPlaceClearProvisionalTemporarily(){
    auto* trackPlaceWnd = static_cast<TrackDesignPlaceWindow*>(window_find_by_class(WC_TRACK_DESIGN_PLACE));
    if (trackPlaceWnd != nullptr){
        trackPlaceWnd->TrackPlaceClearProvisionalTemporarily();
    }
}